controllers/limitador_reconciler.go

```diff
@@ -14,5 +27,85 @@ func NewLimitadorReconciler(client *dynamic.DynamicClient) *LimitadorReconciler
 }
 
 func (r *LimitadorReconciler) Subscription() *controller.Subscription {
-	return &controller.Subscription{}
+	return &controller.Subscription{
+		ReconcileFunc: r.Reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: ptr.To(v1beta1.KuadrantGroupKind), EventType: ptr.To(controller.CreateEvent)},
+			{Kind: ptr.To(v1beta1.LimitadorGroupKind), EventType: ptr.To(controller.DeleteEvent)},
+		},
+	}
+}
+
+func (r *LimitadorReconciler) Reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, _ *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("LimitadorResourceReconciler")
+	logger.Info("reconciling limtador resource", "status", "started")
+	defer logger.Info("reconciling limitador resource", "status", "completed")
+
+	kobjs := lo.FilterMap(topology.Objects().Roots(), func(item machinery.Object, _ int) (*v1beta1.Kuadrant, bool) {
+		if item.GroupVersionKind().Kind == v1beta1.KuadrantGroupKind.Kind {
+			return item.(*v1beta1.Kuadrant), true
+		}
+		return nil, false
+	})
+
+	kobj, err := GetOldestKuadrant(kobjs)
+	if err != nil {
+		if strings.Contains(err.Error(), "empty list passed") {
+			logger.Info("kuadrant resource not found, ignoring", "status", "skipping")
+			return nil
+		}
+		logger.Error(err, "cannont find kuadrant resource", "status", "error")
+		return err
+	}
+
+	aobjs := lo.FilterMap(topology.Objects().Objects().Items(), func(item machinery.Object, _ int) (machinery.Object, bool) {
+		if item.GroupVersionKind().Kind == v1beta1.LimitadorGroupKind.Kind {
+			return item, true
+		}
+		return nil, false
+	})
+
+	if len(aobjs) > 0 {
+		logger.Info("limitador resource already exists, no need to create", "status", "skipping")
+		return nil
+	}
```

Comment:
nit: `aobjs` being authorino objects? So this should be `lobjs` instead? 🤔 
```suggestion
	lobjs := lo.FilterMap(topology.Objects().Objects().Items(), func(item machinery.Object, _ int) (machinery.Object, bool) {
		if item.GroupVersionKind().Kind == v1beta1.LimitadorGroupKind.Kind {
			return item, true
		}
		return nil, false
	})

	if len(lobjs) > 0 {
		logger.Info("limitador resource already exists, no need to create", "status", "skipping")
		return nil
	}
```